### PR TITLE
doggo 0.5.3 (new formula)

### PR DESCRIPTION
**Pull request by vassilevsky**
Homebrew Core has a similar tool (DNS CLI) called `dog` written in Rust:

https://github.com/Homebrew/homebrew-core/blob/master/Formula/dog.rb

This is a more feature-rich tool written in Go.

I am not the author, but a user. I like `doggo`. It's nice.

### CODE REVIEW

**Comment by vassilevsky** [comment]
Not sure how to make PR title better. Followed Formula cookbook.

Not sure what to put into description.

File: Formula/doggo.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    system "go", "build", *std_go_args(ldflags: ldflags), "./cmd/doggo"
+
+    zsh_completion.install "completions/doggo.zsh"
+    fish_completion.install "completions/doggo.fish"
+  end
+
+  test do
+    answer = shell_output("#{bin}/doggo --short example.com NS @1.1.1.1")
+    assert_equal "a.iana-servers.net.\nb.iana-servers.net.\n", answer
+  end
```
Comment (chenrui333) [inline Formula/doggo.rb:28]:
```suggestion

    assert_match version.to_s, shell_output("#{bin}/doggo --version")
  end
```

File: Formula/doggo.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  def install
+    ldflags = [
+      "-s", "-w",
+      "-X", "main.buildVersion=#{version}",
+      "-X", "main.buildDate=2022-06-02"
+    ]
+
+    system "go", "build", *std_go_args(ldflags: ldflags), "./cmd/doggo"
+
+    zsh_completion.install "completions/doggo.zsh"
```
Comment (chenrui333) [inline Formula/doggo.rb:21]:
```suggestion
zsh_completion.install "completions/doggo.zsh" => "_doggo"
```

File: Formula/doggo.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  head "https://github.com/mr-karan/doggo.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = [
+      "-s", "-w",
+      "-X", "main.buildVersion=#{version}",
+      "-X", "main.buildDate=2022-06-02"
+    ]
```
Comment (chenrui333) [inline Formula/doggo.rb:17]:
```suggestion
    ldflags = %W[
      -s -w
      -X main.buildVersion=#{version}
      -X main.buildDate=#{time.iso8601}
    ]
```

File: Formula/doggo.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+class Doggo < Formula
+  desc "Command-line DNS Client for Humans"
+  homepage "https://doggo.mrkaran.dev/"
+  url "https://github.com/mr-karan/doggo.git",
+      tag:      "v0.5.3",
+      revision: "ea7cb3c6cd0467a7db783fbd30aa8599604a76d8"
+  license "GPL-3.0-or-later"
```
Comment (chenrui333) [inline Formula/doggo.rb:7]:
```suggestion
  url "https://github.com/mr-karan/doggo/archive/refs/tags/v0.5.3.tar.gz"
  sha256 "4f8333662fd73c411bf9fb7ce8254d526a777b6491c1d5a49fff0f09b00754c1"
  license "GPL-3.0"
```

Comment (chenrui333) [reply Formula/doggo.rb:7]: need to confirm the license as GPL-3.0 is deprecated.

**Comment by vassilevsky** [comment]
I have applied all suggestions. Thank you!

File: Formula/doggo.rb (hunk 4 above)
Comment (vassilevsky) [reply Formula/doggo.rb]: Tests now fail because of deprecated license spec.

**Comment by chenrui333** [comment]
Thanks @vassilevsky!

**Comment by chenrui333** [comment]
@vassilevsky, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Comment by vassilevsky** [comment]
Thanks for accepting this formula :)

**Comment by mr-karan** [comment]
Thanks for this @vassilevsky @chenrui333 
Excited to see this merged!

**Pull request closed without merging**